build_tools/cmake/riscv.toolchain.cmake

```
@@ -23,14 +23,15 @@ set(RISCV_HOST_TAG linux)
 set(RISCV_TOOL_PATH "$ENV{HOME}/riscv" CACHE PATH "RISC-V tool path")
 
 set(RISCV_TOOLCHAIN_ROOT "${RISCV_TOOL_PATH}/toolchain/clang/${RISCV_HOST_TAG}/RISCV" CACHE PATH "RISC-V compiler path")
+set(RISCV_TOOLCHAIN_PREFIX "" CACHE STRING "RISC-V compiler prefix")
 set(CMAKE_FIND_ROOT_PATH ${RISCV_TOOLCHAIN_ROOT})
 list(APPEND CMAKE_PREFIX_PATH "${RISCV_TOOLCHAIN_ROOT}")
 
-set(CMAKE_C_COMPILER "${RISCV_TOOLCHAIN_ROOT}/bin/clang")
-set(CMAKE_CXX_COMPILER "${RISCV_TOOLCHAIN_ROOT}/bin/clang++")
-set(CMAKE_AR "${RISCV_TOOLCHAIN_ROOT}/bin/llvm-ar")
-set(CMAKE_RANLIB "${RISCV_TOOLCHAIN_ROOT}/bin/llvm-ranlib")
-set(CMAKE_STRIP "${RISCV_TOOLCHAIN_ROOT}/bin/llvm-strip")
+set(CMAKE_C_COMPILER "${RISCV_TOOLCHAIN_ROOT}/bin/${RISCV_TOOLCHAIN_PREFIX}clang")
+set(CMAKE_CXX_COMPILER "${RISCV_TOOLCHAIN_ROOT}/bin/${RISCV_TOOLCHAIN_PREFIX}clang++")
+set(CMAKE_AR "${RISCV_TOOLCHAIN_ROOT}/bin/${RISCV_TOOLCHAIN_PREFIX}ar")
```

```suggestion
set(CMAKE_AR "${RISCV_TOOLCHAIN_ROOT}/bin/${RISCV_TOOLCHAIN_PREFIX}llvm-ar")
```
and same as the tools below

---

If we use the toolsuite built from [riscv-collab/riscv-gnu-toolchain](https://github.com/riscv-collab/riscv-gnu-toolchain) , there is no llvm binutils. Should I add a rule to build llvm binutils in riscv-collab/riscv-gnu-toolchain? Or can we update the toolsuite provided by IREE?

---

IREE's RISCV toolchains were built with  the clang + gcc built flow as discribed in [webdoc](https://openxla.github.io/iree/building-from-source/riscv/#install-risc-v-cross-compile-toolchain-and-emulator), thus they have both gcc and llvm binutils. The `--enable-llvm` flag you pointed in this PR was added last month, while those toolchains were built sometime last year.

Also, IREE's RISCV Linux toolchain has the [multilib](https://github.com/riscv-collab/riscv-gnu-toolchain#build-with-customized-multi-lib-configure) enabled to support both riscv64 and riscv32 builds with the same toolchain.

We could update IREE's RISCV existing toolchains, but unfortunately I don't see a strong reason to do so. If the goal is to add a support to build for riscv64 baremental, the least amount of change seems to be having the proper prefix set for the existing toolchains and the new one. At the same time, @dcaballe can chime in on if we want to formally support/maintain RISCV64 baremetal config, that is, adding the toolchain to the IREE artifact bucket, setting up CI unit test, etc. If not, maybe adding a section in [webdoc source](https://github.com/openxla/iree/blob/main/docs/website/docs/building-from-source/riscv.md) explaining how rv64-baremetal can be supported would be sufficient.

---

Oh...the webdoc describe how to build both gcc and llvm binutils. I'll change back to use llvm binutils. Sorry for not notice this.